### PR TITLE
normalizer: use GRN_CHAR_TYPE

### PR DESCRIPTION
**Pull request by komainu8**
Because char_type is sometimes OR operated with GRN_CHAR_BLANK.

Therefore, when load data that there is whitespace after KATAKANA,
load data are not normalize.
Because KATAKANA before whitespace is operated OR with GRN_CHAR_BLANK,
it can not be decided as KATAKANA.

Fix issues #912

### CODE REVIEW

**Comment by komainu8** [comment]
@kou Could you review this PR, please?

File: lib/normalizer.c [hunk 1]
```diff
@@ -1141,7 +1141,7 @@ grn_nfkc_normalize_unify_stateless(grn_ctx *ctx,
     }
 
     if (data->options->unify_kana &&
-        char_type == GRN_CHAR_KATAKANA &&
+        GRN_CHAR_TYPE(char_type) == GRN_CHAR_KATAKANA &&
         unified_char_length == 3) {
       unifying = grn_nfkc_normalize_unify_kana(unifying, unified_kana);
       if (unifying == unified_kana) {
```
Comment (kou) [inline lib/normalizer.c]: Should we need to change the following line?

```c
char_type = GRN_CHAR_HIRAGANA;
```

```c
char_type = GRN_CHAR_HIRAGANA | (char_type & GRN_CHAR_BLANK));
````

Comment (kou) [reply lib/normalizer.c]: There are more `char_type = ...` in this function.

**Comment by komainu8** [comment]
@kou Thank you for your comments. I've fixed them.

**Comment by kou** [comment]
I've merged but we should also have a test for `test/command/suite/normalizers/nfkc100/`.

**Pull request merged**